### ISSUE TITLE
Training finishes and saves model at step 0

### ISSUE THREAD

**Issue by TheRealBashkim** [emedvedev/attention-ocr#89]
Hi, I'm trying to do training on some reg plates with a training set of 100 images and a test set of 50 images.
When I try and train this is the output given.

2018-05-18 10:23:44,749 root  INFO     phase: train
2018-05-18 10:23:44,749 root  INFO     model_dir: ./checkpoints
2018-05-18 10:23:44,749 root  INFO     load_model: True
2018-05-18 10:23:44,749 root  INFO     output_dir: ./results
2018-05-18 10:23:44,749 root  INFO     steps_per_checkpoint: 100
2018-05-18 10:23:44,749 root  INFO     batch_size: 65
2018-05-18 10:23:44,749 root  INFO     learning_rate: 1.000000
2018-05-18 10:23:44,749 root  INFO     reg_val: 0
2018-05-18 10:23:44,749 root  INFO     max_gradient_norm: 5.000000
2018-05-18 10:23:44,749 root  INFO     clip_gradients: True
2018-05-18 10:23:44,749 root  INFO     max_image_width 160.000000
2018-05-18 10:23:44,749 root  INFO     max_prediction_length 8.000000
2018-05-18 10:23:44,749 root  INFO     channels: 1
2018-05-18 10:23:44,749 root  INFO     target_embedding_size: 10.000000
2018-05-18 10:23:44,749 root  INFO     attn_num_hidden: 128
2018-05-18 10:23:44,759 root  INFO     attn_num_layers: 2
2018-05-18 10:23:44,759 root  INFO     visualize: False
2018-05-18 10:23:55,431 root  INFO     Reading model parameters from ./checkpoints\model.ckpt-0
2018-05-18 10:23:55,984 root  INFO     num_epoch: 1000
2018-05-18 10:23:57,739 root  INFO     Starting the training process.
2018-05-18 10:24:35,500 root  INFO     Global step 0. Time: 0.000, loss: 0.000000, perplexity: 1.00.
2018-05-18 10:24:35,500 root  INFO     Finishing the training and saving the model at step 0.

and when I am testing

2018-05-18 10:25:15,129 root  INFO     phase: test
2018-05-18 10:25:15,129 root  INFO     model_dir: ./checkpoints
2018-05-18 10:25:15,129 root  INFO     load_model: True
2018-05-18 10:25:15,129 root  INFO     output_dir: ./results
2018-05-18 10:25:15,129 root  INFO     steps_per_checkpoint: 0
2018-05-18 10:25:15,129 root  INFO     batch_size: 1
2018-05-18 10:25:15,129 root  INFO     learning_rate: 1.000000
2018-05-18 10:25:15,129 root  INFO     reg_val: 0
2018-05-18 10:25:15,129 root  INFO     max_gradient_norm: 5.000000
2018-05-18 10:25:15,129 root  INFO     clip_gradients: True
2018-05-18 10:25:15,129 root  INFO     max_image_width 160.000000
2018-05-18 10:25:15,129 root  INFO     max_prediction_length 8.000000
2018-05-18 10:25:15,129 root  INFO     channels: 1
2018-05-18 10:25:15,129 root  INFO     target_embedding_size: 10.000000
2018-05-18 10:25:15,129 root  INFO     attn_num_hidden: 128
2018-05-18 10:25:15,129 root  INFO     attn_num_layers: 2
2018-05-18 10:25:15,129 root  INFO     visualize: False
2018-05-18 10:25:17,521 root  INFO     Reading model parameters from ./checkpoints\model.ckpt-0

Any help would be greatly appreciated.
Thanks :)

**Comment by emedvedev** [comment]
It looks like your training set might be smaller than a default batch size. Are you sure it's 100 and not less (the batch size is 65)? Try setting the batch size to 10 or 20 and see if it's any different.

On a related note, 100 images sounds like a very small set to me. Not sure whether you'd get any kind of results with a set of this size.

**Comment by TheRealBashkim** [comment]
Thanks for the reply, I've tried setting the batch size to 10 and 20 with no change.
I will try it on a bigger set, how many images would you recommend for a decent set size.

**Comment by emedvedev** [comment]
It depends on your particular use case, but I would say you should start with a few thousands.

Try to remove the `checkpoints` dir before starting, too. Also, which version do you use and did you install it from pip or clone the `master` branch?

**Comment by emedvedev** [comment]
Please let me know if you still have the same issue on a larger dataset starting from scratch (i.e. with the `checkpoints` dir removed). You could also try installing from pip through `pip install aocr`, but it's unlikely to give different results, since it's more or less the same version.

I'll try testing with a small dataset, too, but it's unlikely I can get to it until the end of the month, sadly.

**Comment by adityathakker** [comment]
Hey @TheRealBashkim!
I had the same problem before. Turns out, it was happening because all my images had a width greater than the **max_width** specified. That's why it was getting stuck in a While loop in the **gen** function in **data_gen.py**.
